apis/rust/node/src/event_stream/mod.rs

```diff
@@ -104,11 +108,26 @@ impl EventStream {
 
     /// wait for the next event on the events stream.
     pub fn recv(&mut self) -> Option<Event> {
-        futures::executor::block_on(self.recv_async())
+        futures::executor::block_on(self.recv_async(None))
     }
 
-    pub async fn recv_async(&mut self) -> Option<Event> {
```

Comment:
Changing the signature of `recv_async` would be a breaking change. Could we add a new `recv_async_timeout` method instead?

Reply:
Sure make sense!